server/repository/src/db_diesel/key_value_store.rs

```diff
@@ -39,7 +42,13 @@ pub enum KeyValueType {
     SettingsTokenSecret,
 }
 
-#[derive(Clone, Queryable, Insertable, AsChangeset, Debug, PartialEq)]
+impl Default for KeyValueType {
```

Comment:
Need to set sync site in test, which requires KeyValueStoreRow being generated

Reply:
Maybe this could be a comment in the code rather than just in the PR?

Reply:
I think we would need to set other KeyValue rows in test data in the future. Don't want the comment to be specific to sync site in test. I just wanted to tell reviewer why it's added in this particular PR.

Reply:
Make sense. I just find it weird having a Default implementation without an obvious default value :)

Reply:
Hmm, but have to set a value for it, so just chose one (there are a few places already where this is done with defaults that don't really make sense). 

Another hindsight, instead of inline_init method, implement a few methods on the data type, so we can chain inline declaration:

```rust
impl InvoiceRow {
   fn new(/* all required for blank insertion to db without reasonable defaults, so FK constraint and type */ ) -> Self {}
   fn set_{fieldname}(mut self) -> Self {} // a bit tideous but much nicer to consume, see below 
}
// previous
inline_init(|r: &mut InvoiceRow| {
    r.id = "invoice1".to_string();
    r.name_id = name_1().id.to_string();
    r.store_id = store_1().id.to_string();
    r.r#type = InvoiceRowType::InboundShipment;
    r.status = InvoiceRowStatus::New;
    r.invoice_number = 12;
})
// new
InvoiceRow::new(
    "invoice1",
    &name_1().id,
    &store_1().id,
    InvoiceRowType::InboundShipment,
)
.set_status(InvoiceRowStatus::New)
.set_invoice_number(12);
// and KeyValueRow
KeyValueRow::new(KeyValueType::set_i32).set_value_int(mock_store_b().site_id);
```

The main reason inline_init was chosen, we already had a mass of mock data, and a [script was made](https://github.com/openmsupply/open-msupply/blob/develop/server/util/src/to_inline_init.js) to auto convert to inline init for a file/type

Reply:
Yeah, all good, I'll get over it :)
Just thought it was worth a comment, just don't like the idea of someone relying on a default value, but I think we can pick that up in future code reviews if anyone was doing this.

Definitely not a big deal, and no point to waste time on it.